src/snovault/schema_views.py

```diff
@@ -65,3 +69,30 @@ def schemas(context, request):
         subtypes[name] = type_info.subtypes
 
     return schemas
+
+
+@view_config(route_name='schemas_map', request_method='GET',
+             decorator=etag_app_version_effective_principals)
+def schemas_map(context, request):
+    """Returns a mapping of schema object name to the corresponding schema ID.
+
+    This gets called as a response to a GET request for a schema map.
+    The `schemas` function returns this information as well, but as it includes
+    all schema definitions, this response becomes quite large. This function
+    only assembles and returns a dict keyed by each schema object name (e.g.
+    "GeneticModification") with a value of the corresponding schema ID (in this
+    example that would be "/profiles/genetic_modification.json"). Any schemas
+    without an `id` property don't get included.
+
+    To direct this response to a front-end React component, the returned JSON
+    also includes an @type of 'JSONSchemasMap'.
+    """
+    types = request.registry[TYPES]
+    map = {}
```

Comment:
map is a built in function so we can't use it as a variable.

Reply:
OK; didn’t know about that.